run_route_scripts/gen_requests.py

```diff
@@ -22,20 +25,42 @@ def get_random_locs(yc, xc, radius, count):
         y = r * math.sin(c)
         x += xc
         y += yc
-        locs.append({'lon':round(x, 6),'lat':round(y, 6)});
+        locs.append({'lon':round(x, 6),'lat':round(y, 6),'type':"break"});
     return locs
+    
+def get_random_intermediate_locs(yc, xc, radius, count, intermediate):
+    intermediate_locs = []
+    rads = radius / 110567.0
+    for i in range(0, count):
+        a = float(random.uniform(0.0, 1.0))
+        b = float(random.uniform(0.0, 1.0))
+        r = rads * math.sqrt(a)
+        c = 2 * math.pi * b
+        x = r * math.cos(c)
+        y = r * math.sin(c)
+        x += xc
+        y += yc
+        intermediate_locs.append({'lon':round(x, 6),'lat':round(y, 6),'type':intermediate});
+    return intermediate_locs
 
 
 # do each input quadruplet
 locs = []
+intermediate_locs = []
 for arg in sys.argv[1:]:
-    y, x, radius, count = arg.split(',')
+    y, x, radius, count, intermediate = arg.split(',')
     locs.extend(get_random_locs(float(y), float(x), float(radius), int(count)))
+    intermediate_locs.extend(get_random_intermediate_locs(float(y), float(x), float(radius), int(count), str(intermediate)))
 
 # make requests
 for start in locs:
-    for end in locs:
+  for intermediates in intermediate_locs:
+     for end in locs:
```

Comment:
seems like this should be unnested and then intermediates may or may not produce a list of random intermediate locations